arrow-data/src/equal/mod.rs

```diff
@@ -96,6 +96,7 @@ fn equal_values(
             variable_sized_equal::<i64>(lhs, rhs, lhs_start, rhs_start, len)
         }
         DataType::FixedSizeBinary(_) => fixed_binary_equal(lhs, rhs, lhs_start, rhs_start, len),
+        DataType::BinaryView | DataType::Utf8View => unimplemented!("Not implemented"),
```

Comment:
It might be nice to add a little more context to the errors 

```suggestion
        DataType::BinaryView | DataType::Utf8View => unimplemented!("BinaryView/Utf8View not yet implemented"),
```